spock-guice/guice.gradle

```diff
@@ -6,7 +6,8 @@ description = "Spock's Guice Module provides support for testing Guice 2/3 based
 
 dependencies {
   api project(":spock-core")
-  implementation "com.google.inject:guice:3.0", provided
+  compileOnly "com.google.inject:guice:3.0"
+  testImplementation "com.google.inject:guice:3.0"
```

Comment:
Shouldn't these (and below for the other specific modules) be normal `implementation` dependencies? Consumers could still use different versions by using Gradle's dependency management capabilities.

Reply:
They had been marked as `provided` in Maven, which basically makes them non-transitive. This PR just aligns the behavior between Gradle and Maven.